### ISSUE TITLE
BFA Update Error

### ISSUE THREAD

**Issue by VaporAPX** [AdiAddons/AdiButtonAuras#275]
Logged into the beta tonight after the recent update, got the following error when logging into my paladin.

> 8x ...Ons\Ace3\CallbackHandler-1.0\CallbackHandler-1.0-6.lua:145: Invalid spell identifier: 213757
[C]: ?
AdiButtonAuras\core\RuleDSL.lua:98: in function <AdiButtonAuras\core\RuleDSL.lua:93>
AdiButtonAuras\core\RuleDSL.lua:126: in function <AdiButtonAuras\core\RuleDSL.lua:125>
AdiButtonAuras\core\RuleDSL.lua:197: in function <AdiButtonAuras\core\RuleDSL.lua:196>
[C]: ?
AdiButtonAuras\core\Core.lua:389: in function `?'
...Ons\Ace3\CallbackHandler-1.0\CallbackHandler-1.0-6.lua:145: in function <...Ons\Ace3\CallbackHandler-1.0\CallbackHandler-1.0.lua:145>
[string "safecall Dispatcher[1]"]:4: in function <[string "safecall Dispatcher[1]"]:4>
[C]: ?
[string "safecall Dispatcher[1]"]:13: in function `?'
...Ons\Ace3\CallbackHandler-1.0\CallbackHandler-1.0-6.lua:90: in function `Fire'
...ttonAuras\libs\LibSpellbook-1.0\LibSpellbook-1.0-21.lua:378: in function <...ttonAuras\libs\LibSpellbook-1.0\LibSpellbook-1.0.lua:338>
(tail call): ?
Locals:
nil

**Comment by Rainrider** [comment]
I am still updating dependent libraries for the moment. Class data is another story and I'll open another issue to track progress on this front. I will need help that though as class data is quite time consuming (about an hour per spec depending on your class knowledge and how comfortable you are editing LibPlayerSpells). If you would like to help with that, you might read [the contribution guide for LPS](https://github.com/AdiAddons/LibPlayerSpells-1.0/wiki/Contribution-Guidelines).

I will also have to change a thing or two development wise if we want to keep distributing a disembedded version.

TL;DR
A lot of work has to be done and if no one helps with class data, I won't be able to get it updated for all classes before BfA launch.

**Comment by VaporAPX** [comment]
I'll do my best to assist, I have basic knowledge of LUA and editing the player spells portion isn't all that difficult from what I can tell, just very time consuming like you said.  I've never done any official updating on github so that'll be a bit of a learning curve as I don't want to cause more problems than I'm fixing if I help out lol.  Reading through the contribution guide as we speak.

**Comment by VaporAPX** [comment]
So for this particular problem, they changed Execution Sentence from being a ticking dot to a "increases holy damage taken" debuff.  I've gone into the Paladin sections in both the rules and data sections and changed the spell ID's, but I can't get the action button to light red.  No matter how I manipulate the spell ID's it won't light red, so I'm not sure what else I'm missing.  

(Great start for tryin to help you update this lol)

**Comment by VaporAPX** [comment]
@Rainrider  Just looking for a quick bit of info from you.  I forked a version of LibPlayerSpells and did the updates to Execution Sentence that were causing issues under a new branch called BFA.  My question is, how do I get my fork to merge with the original.  I can't find anywhere to make a BFA branch on the original and I don't want to merge mine with the master.  Just looking for a quick clarification before I do anything, don't wanna mess anything up lol.

**Comment by Rainrider** [comment]
Thank you very much, I really appreciate it.

You were a bit too fast though :) I just pushed a `bfa` branch and it should be the base branch for BfA development. I also pushed my changes for marksmanship hunter in a separate branch named `bfa-hunter` and issued a pull request (PR) against the `bfa` branch.

My workflow for class data for a new expansion is like that:
1. create a topic branch off the development branch (bfa in this case):  
  `git fetch`
  `git checkout bfa`
  `git checkout -b bfa-hunter` this creates the bfa-hunter branch off the bfa branch
2. remove all spells from the class file to get a clean starting point
3. add base spells for the spec (I started with Marksmanship hunter) and commit
4. add the talents for the spec and commit
5. add the pvp talents for the spec (this will have to wait until LibSpellbook is updated)
6. repeat for the other specs

You can create a pull request (PR) against the bfa branch after point 3. You can add further commits to your branch and they will automatically appear in the PR too.

I start with an empty class file because often spells are still in-game but are not related to the class anymore. You won't get errors for those so they are easy to miss. Also I know I reviewed all the spells I added and not just forgot something that was still handing around in the file.

Please commit on every stage (base, talents) and not on every spell you add. It makes it easier to review and test your changes. Ofc if you forgot a spell or something, you can add a separate commit for that, but don't let that be the rule.

As for how to issue pull requests, please read https://help.github.com/articles/creating-a-pull-request/

If it is ok for you, it would be nice if you could re-apply your changes following this workflow.

**Comment by VaporAPX** [comment]
I have no problem following that workflow if that's the preferred method, absolutely.  The only thing I'm having trouble doing atm is creating the branch, in this case "bfa-paladin".  Do I need user access to be able to assist in this?  I made a mock up repository to do my own testing and I can create branches with no issue, so I'm not sure if it's me noobing out hard or if I need to be allowed access.

Apologies for the questions, I'm decent with muddling in lua, but github is uncharted territory for me.  Thank you for all the assistance in this, once I get it figured out I'll actually be helpful I promise lol.

**Comment by Rainrider** [comment]
Follow [this guide](https://help.github.com/articles/fork-a-repo/) for forking LibPlayerSpells (LPS). I think you figured that one out already since you have your fork.

https://help.github.com/articles/creating-a-pull-request-from-a-fork/ tells how to issue the pull request from the fork. You don't need special rights on upstream, you just have to have pushed your changes to your fork first.

Oh, do `git fetch upstream` instead of `git fetch` (the latter will fetch from your fork rather than from upstream, sorry for that, I'll update my comment above for future reference)

**Comment by Rainrider** [comment]
I'll try to put up a guide for contributing to ABA since setting up the whole thing is a bit of a hassle. You will need some of the dependencies (most notably [LibSpellbook](https://github.com/AdiAddons/LibSpellbook-1.0) besides LPS) in order to test your changes.

**Comment by VaporAPX** [comment]
I think where I'm going wrong is I'm not using git itself, really really limited knowledge there, and have been only using github itself to do everything.  My night shift is over, so after I get a few hours sleep I'll try this again lol.  Thank you again.

**Comment by Rainrider** [comment]
See https://github.com/AdiAddons/AdiButtonAuras/wiki/Contributing#setting-up-dependencies.
I'll try to have LibSpellbook-1.0 updates done today, you will have to checkout the bfa branch there as well. You only need this for testing the pvp talents through ABA though, so you don't have to wait for that.

Sorry that the whole thing is so interdependent. I know it is a hassle, but it is like that for the time being.

**Comment by Rainrider** [comment]
LibSpellbook has pvp talents support now. I actually ended up removing it as it seems honor talents are in the spellbook when they are available.

I'll have to see how about azerite traits there.
